src/main/java/io/split/android/client/service/sseclient/sseclient/SseHandler.java

```diff
@@ -74,8 +83,11 @@ public void handleError(boolean retryable) {
 
     private void handleControlNotification(IncomingNotification incomingNotification) {
         try {
-            mNotificationManagerKeeper.handleControlNotification(
-                    mNotificationParser.parseControl(incomingNotification.getJsonData()));
+            ControlNotification notification = mNotificationParser.parseControl(incomingNotification.getJsonData());
+            notification.setTimestamp(incomingNotification.getTimestamp());
+            // If push disabled also set isPaused = true to avoid process any notification until push shuts down
+            isStreamingPaused.set(!(notification.getControlType() == ControlNotification.ControlType.STREAMING_ENABLED));
```

Comment:
shouldn't we also consider OCCUPANCY notifications to set if streaming is paused or not?

if that is the case, you could consider having a synchronous `isStreamingPaused` method in the NotificationKeeper class instead of an AtomicBoolean in SseHandler.

Reply:
Perfect, also updated.